akka-testkit-typed/src/main/scala/akka/testkit/typed/internal/Within.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.testkit.typed.internal
+
+import akka.annotation.InternalApi
+import akka.testkit.typed.TestKitSettings
+
+import scala.concurrent.duration._
+import akka.util.PrettyDuration._
+import akka.testkit.typed.scaladsl._
+
+@InternalApi
+private[akka] trait Within {
+
+  implicit def settings: TestKitSettings
+
+  private var end: Duration = Duration.Undefined
+
+  /**
+   * if last assertion was expectNoMessage, disable timing failure upon within()
+   * block end.
+   */
+  private var finishedProcessing = false
+
+  def setFinished(isFinished: Boolean) = finishedProcessing = isFinished
```

Comment:
does this have to be public?

Reply:
I think so... The Within somehow needs to know when the job is finished.
In TestProbe it was checking for messages and setting an internal variable when it had finished, by making this more generic we can not rely on the same mechanism, can we?

Reply:
when would users need this?
I have never used it.
If public it must be documented.

Reply:
I'm sorry you're right, this is only used by the implementations of Within.